airflow/callbacks/callback_requests.py

```diff
@@ -74,8 +75,8 @@ def __init__(
         self.is_failure_callback = is_failure_callback
 
     def to_json(self) -> str:
-        dict_obj = self.__dict__.copy()
-        dict_obj["simple_task_instance"] = dict_obj["simple_task_instance"].__dict__
+        dict_obj = copy.deepcopy(self.__dict__)
```

Comment:
Was there a reason we changed this to deepcopy? It feels like it shouldn't be required if we are also calling `simple_ti.as_dict()`

Reply:
I changed to `self.simple_task_instance.as_dict()` and it solved the issue I was having before that made me use deepcopy. Previously with just copy, my test was not passing as input != result.
Thanks